src/main/java/com/onevizion/scmdb/DbManager.java

```diff
@@ -258,4 +255,50 @@ public void generateDdlForAllObjects() {
         ddlGenerator.executeSettingTransformParams();
         ddlGenerator.generateDllsForAllDbObjects();
     }
+
+    /**
+     * The method filters scripts by type and sorts them in the correct execution order.
+     * Scripts are divided into two list with low priority and high priority, all scripts in the range 1..100
+     * fall into the list with low priority and must be executed last.
+     * As a result, the groups are assembled into a list in turn, starting with a high priority.
+     * If the type is ROOLBACK, then each group will be sorted in reverse order and as a result,
+     * the groups are collected in a list in turn, starting with the lowest priority.
+     *
+     * Example 1 : Input  scripts = [1_script1.sql, 1_script1_rollback.sql, 8861_script2.sql, 8861_script2_rollback.sql]
+     *                    scriptType = COMMIT
+     *             Output List<SqlScript> = [8861_script2.sql, 1_script1.sql]
+     *
+     * Example 2 : Input  scripts = [1_script1.sql, 1_script1_rollback.sql, 8861_script2.sql, 8861_script2_rollback.sql]
+     *                    scriptType = ROLLBACK
+     *             Output List<SqlScript> = [1_script1_rollback.sql, 8861_script2_rollback.sql]
+     *
+     * @param scripts list of scripts to sort
+     * @param scriptType script type
+     * @return list of scripts with the correct execution order
+     */
+    private List<SqlScript> sortScriptsInExecutionOrder(List<SqlScript> scripts, ScriptType scriptType) {
+        scripts = scripts.stream()
+                         .filter(script -> script.getType() == scriptType)
+                         .collect(Collectors.toList());
+
+        List<SqlScript> highPriorityList = new ArrayList<>();
+        List<SqlScript> lowPriorityList = new ArrayList<>();
+        for (SqlScript script : scripts) {
+            if (script.getName().matches(SCRIPT_NUMBERING_IS_MORE_THAN_THREE_DIGITS_REGEX)) {
+                highPriorityList.add(script);
+            } else {
+                lowPriorityList.add(script);
+            }
+        }
+
+        if (scriptType == ROLLBACK) {
+            highPriorityList.sort(Comparator.reverseOrder());
+            lowPriorityList.sort(Comparator.reverseOrder());
+            lowPriorityList.addAll(highPriorityList);
+            return lowPriorityList;
+        } else {
+            highPriorityList.addAll(lowPriorityList);
```

Comment:
you can't be sure that `highPriorityList` and `lowPriorityList` are sorted properly, add explicit sort here

Reply:
Added.